lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runime/azbatch/util/AbstractCommandBuilder.java

```diff
@@ -72,13 +69,19 @@ public String build(final JobSubmissionEvent jobSubmissionEvent) {
   private List<String> getCommandList(
       final JobSubmissionEvent jobSubmissionEvent) {
 
-    return new JavaLaunchCommandBuilder(this.launcherClass, this.commandListPrefix, this.classpathSeparatorChar)
+    return new JavaLaunchCommandBuilder(this.launcherClass, this.commandListPrefix)
         .setJavaPath("java")
         .setConfigurationFilePaths(Collections.singletonList(this.reefFileNames.getDriverConfigurationPath()))
-        .setClassPath(this.classpathProvider.getDriverClasspath())
+        .setClassPath(getDriverClasspath())
         .setMemory(jobSubmissionEvent.getDriverMemory().get())
         .setStandardOut(STD_OUT_FILE)
         .setStandardErr(STD_ERR_FILE)
         .build();
   }
+
+  /**
+   * Returns the driver classpath string which is compatible with the intricacies of the OS.
+   * @return classpath parameter string.
+   */
+  protected abstract String getDriverClasspath();
```

Comment:
Please add a task to review/complete Javadocs after all the code is completed. You've done a good job putting in comments but they're a little inconsistent.

Reply:
sounds good.